internal/service/streaminstance/model_stream_instance.go

```diff
@@ -9,7 +9,7 @@ import (
 	"go.mongodb.org/atlas-sdk/v20231115002/admin"
 )
 
-func NewStreamInstanceCreateReq(ctx context.Context, plan *TFStreamInstanceRSModel) (*admin.StreamsTenant, diag.Diagnostics) {
+func NewStreamInstanceCreateReq(ctx context.Context, plan *TFStreamInstanceModel) (*admin.StreamsTenant, diag.Diagnostics) {
```

Comment:
don't we need to differentiate between RS and DS model?

Reply:
not for this case as attributes are the same (ideal situation)

Reply:
should we still keep consistent though? I don't have a strong opinion, I am fine with both, but just keep in mind that if it happened in other resources that it had to be different, it might as well happen here.

Reply:
If the attributes of RS and DS are identical I like the benefit of reusing the model as it ensures alignment. If we intentionally need to add an attribute to only one of the 2 we can define the separate models in that change.